File: benchmarks/suite5/reduce.rs
mod bench_runtime;
use bench_runtime::suite5_print_i64;

fn main() {
    let n: i64 = 10_000_000;
    let mut acc: i64 = 0;
    let mut i: i64 = 0;
    while i < n {
        acc = acc + i * 31 - i / 8 + i % 13;
        i += 1;
    }
    suite5_print_i64(acc);
}

File: benchmarks/suite5/reduce.zig
const bench = @import("bench.zig");
const clib = @cImport({
    @cInclude("stdio.h");
});

pub fn main() void {
    const n: i64 = bench.opaqueI64(10000000);
    var acc: i64 = 0;
    var i: i64 = 0;
    while (i < n) : (i += 1) {
        acc = acc + i * 31 - @divTrunc(i, 8) + @rem(i, 13);
    }
    bench.printI64(acc);
}
